crypto/bn/bn_shift.c

```diff
@@ -173,3 +197,47 @@ int BN_rshift(BIGNUM *r, const BIGNUM *a, int n)
     bn_check_top(r);
     return 1;
 }
+
+int bn_rshift_fixed_top(BIGNUM *r, const BIGNUM *a, int n)
+{
+    int i, top, nw, lb, rb;
+    BN_ULONG *t, *f;
+    BN_ULONG l, m, mask;
+
+    bn_check_top(r);
+    bn_check_top(a);
+
+    assert(n >= 0);
+
+    nw = n / BN_BITS2;
+    if (nw >= a->top || a->top == 0) {
```

Comment:
Is the second conditional here actually necessary? Isn't it always caught by the first?

Reply:
Again this also implies `n` is public, so we should probably mention that.